src/utils/system_utils/tests/system_utils_ut.cpp

```diff
@@ -9,81 +9,214 @@
 using Catch::Matchers::Equals;
 
 #include "aduc/system_utils.h"
-
 #include <aduc/auto_opendir.hpp>
-#include <algorithm> // std::find*
-#include <dirent.h>
-#include <sstream>
 #include <sys/stat.h>
 #include <vector>
 
-static bool callback_called = false;
-std::string actual;
+/**
+ * @brief A capture of the arguments passed in a call invocation to callback for each sub dir in base dir.
+ *
+ */
+struct ForEachDirCallRecord
+{
+    void* context;
```

Comment:
see my comment below -- this might be better as a concrete object (or typed object pointer) rather than void* as you *know* what type this will be in this test.  if you're trying to do something reusable (which is good), maybe make this templated, but that latter is overkill for a UT

Reply:
created bug to capture these 2 remaining comments.

Reply:
Fixed in PR: https://github.com/Azure/iot-hub-device-update/pull/308